lib/publisher.ex

```diff
@@ -117,7 +123,7 @@ defmodule GenRMQ.Publisher do
 
   @doc false
   def handle_call({:publish, msg, key, headers}, _from, %{channel: channel, config: config} = state) do
-    metadata = config |> base_metadata |> Keyword.merge(headers: headers)
+    metadata = config |> base_metadata() |> merge_metadata(headers)
```

Comment:
@soundmonster let's rename `headers` -> `metadata`

Reply:
👍

Reply:
This is a potentially breaking change though, so I would advise to bump the minor version on this one.